Modules/Net/readers.swift

```diff
@@ -93,22 +98,100 @@ internal class UsageReader: Reader<Network_Usage> {
         }
         freeifaddrs(interfaceAddresses)
         
-        if self.usage.upload != 0 && self.usage.download != 0 {
-            self.usage.upload = upload - self.usage.upload
-            self.usage.download = download - self.usage.download
+        if !verifiedUsageDataApproach {
+            if didNativeApproachReportNonZeroDownload || nonZeroUpdatesCount >= 3 {
+                if !didNativeApproachReportNonZeroDownload {
+                    shouldUseProcessesUsageData = true
+                }
+                verifiedUsageDataApproach = true
+            } else {
+                let processesUsage = allProcessesTotalUsage()
+                let accumulatedDownload = processesUsage?.download ?? 0
+                if download > 0 || accumulatedDownload > 0 {
+                    nonZeroUpdatesCount += 1
+                }
+                if download > 0 {
+                    didNativeApproachReportNonZeroDownload = true
+                }
+            }
         }
         
-        if self.usage.upload < 0 {
-            self.usage.upload = 0
-        }
-        if self.usage.download < 0 {
-            self.usage.download = 0
+        let accumulatedDownload: Int64
+        if shouldUseProcessesUsageData {
+            let processesUsage = allProcessesTotalUsage()
+            accumulatedDownload = processesUsage?.download ?? 0
+        } else {
+            accumulatedDownload = download
         }
```

Comment:
I did not really understand why there are so many checks. But it still not checking properly.

Correct me please if I miss something.
`nonZeroUpdatesCount` checks if the last 3 reads were 0. But it normal that in the last second traffic was 0. And even more than the last 3 seconds. 

Also, this approach does not work if I switch to some interface that does not have traffic. And now it starts to showing some download values. And it can be like this.